components/cli/ballerina/docker_bal_executor.go

```diff
@@ -137,12 +139,101 @@ func (baleExecutor *DockerBalExecutor) Build(fileName string, iName []byte) erro
 	return nil
 }
 
+// Run executes ballerina run when ballerina is not installed.
+func (balExecutor *DockerBalExecutor) Run(balFile, imageDir string, args []string, envVars []EnvironmentVariable) error {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error in determining working directory, %v", err)
+	}
+	//Retrieve the cellery cli docker instance status.
+	cmdDockerPs := exec.Command("docker", "ps", "--filter", "label=ballerina-runtime="+version.BuildVersion(),
+		"--filter", "label=currentDir="+currentDir, "--filter", "status=running", "--format", "{{.ID}}")
+
+	containerId, err := cmdDockerPs.Output()
+	if err != nil {
+		return fmt.Errorf("docker Run Error, %v", err)
+	}
+
+	if string(containerId) == "" {
+		cmdDockerRun := exec.Command("docker", "run", "-d", "-l", "ballerina-runtime="+version.BuildVersion(),
+			"--mount", "type=bind,source="+util.UserHomeDir()+string(os.PathSeparator)+".ballerina,target=/home/cellery/.ballerina",
+			"--mount", "type=bind,source="+util.UserHomeDir()+string(os.PathSeparator)+".cellery,target=/home/cellery/.cellery",
+			"--mount", "type=bind,source="+util.UserHomeDir()+string(os.PathSeparator)+".kube,target=/home/cellery/.kube",
+			"wso2cellery/ballerina-runtime:"+version.BuildVersion(), "sleep", "600",
+		)
+		containerId, err = cmdDockerRun.Output()
+		if err != nil {
+			return fmt.Errorf("docker Run Error, %v", err)
+		}
+		time.Sleep(5 * time.Second)
+	}
+	cliUser, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("error while retrieving the current user, %v", err)
+	}
+	exeUid := dockerCliUserId
+
+	if cliUser.Uid != dockerCliUserId && runtime.GOOS == "linux" {
+		cmdUserExist := exec.Command("docker", "exec", strings.TrimSpace(string(containerId)),
+			"id", "-u", cliUser.Username)
+		_, errUserExist := cmdUserExist.Output()
+		if errUserExist != nil {
+			cmdUserAdd := exec.Command("docker", "exec", strings.TrimSpace(string(containerId)), "useradd", "-m",
+				"-d", "/home/cellery", "--uid", cliUser.Uid, cliUser.Username)
+
+			_, errUserAdd := cmdUserAdd.Output()
+			if errUserAdd != nil {
+				return fmt.Errorf("error in adding Cellery execution user, %v", err)
+			}
+		}
+		exeUid = cliUser.Uid
+	}
+	re := regexp.MustCompile(`^.*cellery-cell-image`)
+	balFile = re.ReplaceAllString(balFile, "/home/cellery/.cellery/tmp/cellery-cell-image")
```

Comment:
move to a constant and refer from all hard coded places.

Reply:
Updated.